notification/job.go

```diff
@@ -303,6 +303,187 @@ func ProcessPendingGroupedNotifications(parentCtx context.Context) error {
 	return nil
 }
 
+func ProcessFallbackCheckNotificationsJob(ctx context.Context) *job.Job {
+	return &job.Job{
+		Name:       "ProcessFallbackCheckNotifications",
+		Retention:  job.RetentionFailed,
+		JobHistory: true,
+		RunNow:     true,
+		Context:    ctx,
+		Singleton:  false,
+		Schedule:   "@every 30s",
+		Fn: func(ctx job.JobRuntime) error {
+			var errCount int
+			for {
+				done, err := ProcessFallbackCheckNotifications(ctx.Context)
+				if err != nil {
+					ctx.History.AddErrorf("failed to process notifications in status=%s : %v", models.NotificationStatusCheckingFallback, err)
+
+					errCount++
+					if errCount > 3 {
+						// avoid getting stuck in a loop
+						break
+					}
+
+					time.Sleep(2 * time.Second) // prevent spinning on db errors
+					continue
+				}
+
+				ctx.History.IncrSuccess()
+
+				if done {
+					break
+				}
+			}
+
+			return nil
+		},
+	}
+}
+
+func ProcessFallbackCheckNotifications(parentCtx context.Context) (bool, error) {
```

Comment:
Why do we need jobs for this? Can't we do it inline

Reply:
https://github.com/flanksource/duty/pull/1347